Source/Csla.AspNetCore/ApplicationContextManagerHttpContext.cs

```diff
@@ -0,0 +1,126 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ApplicationContextManager.cs" company="Marimer LLC">
+//     Copyright (c) Marimer LLC. All rights reserved.
+//     Website: https://cslanet.com
+// </copyright>
+// <summary>Application context manager that uses HttpContextAccessor</summary>
+//-----------------------------------------------------------------------
+using Csla.Core;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Security.Claims;
+
+namespace Csla.AspNetCore
+{
+  /// <summary>
+  /// Application context manager that uses HttpContextAccessor when 
+  /// resolving HttpContext to store context values.
+  /// </summary>
+  public class ApplicationContextManagerHttpContext : IContextManager
+  {
+
+    private const string _localContextName = "Csla.LocalContext";
+    private const string _clientContextName = "Csla.ClientContext";
+
+    /// <summary>
+    /// Creates an instance of the object, initializing it
+    /// with the required IServiceProvider.
+    /// </summary>
+    /// <param name="httpContextAccessor">HttpContext accessor</param>
+    public ApplicationContextManagerHttpContext(IHttpContextAccessor httpContextAccessor)
+    {
+      HttpContext = httpContextAccessor.HttpContext;
+    }
+
+    /// <summary>
+    /// Gets the current HttpContext instance.
+    /// </summary>
+    protected virtual HttpContext HttpContext { get; private set; }
+
+    /// <summary>
+    /// Gets a value indicating whether this
+    /// context manager is valid for use in
+    /// the current environment.
+    /// </summary>
+    public bool IsValid
+    {
+      get { return HttpContext != null; }
+    }
+
+    /// <summary>
+    /// Gets the current principal.
+    /// </summary>
+    public System.Security.Principal.IPrincipal GetUser()
+    {
+      var result = HttpContext?.User;
+      if (result == null)
+      {
+        result = new Csla.Security.CslaClaimsPrincipal();
+        SetUser(result);
+      }
+      return result;
+    }
+
+    /// <summary>
+    /// Sets the current principal.
+    /// </summary>
+    /// <param name="principal">Principal object.</param>
+    public void SetUser(System.Security.Principal.IPrincipal principal)
+    {
+      HttpContext.User = (ClaimsPrincipal)principal;
```

Comment:
This doesn't handle the possibility of a null HttpContext like the other code. I can't decide if that would be relevant by this point. Would this already have crashed before that? I don't _think_ so, so maybe we should handle the possibility?

Reply:
"Handle" would mean throwing an exception. Which could be valid, as we could throw something more instructive than a generic null reference exception maybe.

However, this really is a null reference scenario.